### PR TITLE
feat: add the optional proxy parameter for pjsua

### PR DESCRIPTION
**Pull request by emrahcom**
This commit adds an optional `--proxy` parameter for `pjsua` client. It allows to set  SIP `proxy` through `jitsi-component-selector` API.

If there is no `proxy` in the API request, the application still sets the forced proxy to not break the old setups.
`--proxy=...transport=tcp;hide`

A sample request may be like the following

```json
{
  "callParams": {
    "callUrlInfo": {
      "baseUrl": "$JITSI_HOST",
      "callName": "$JITSI_ROOM"
    }
  },
  "componentParams": {
    "type": "SIP-JIBRI",
    "region": "default-region",
    "environment": "default-env"
  },
  "metadata": {
    "sipClientParams": {
      "userName": "$CALLER_USERNAME",
      "password": "$CALLER_PASSWORD",
      "sipAddress": "$CALLEE",
      "displayName": "Caller",
      "proxy": "$SIP_PROXY",
      "autoAnswer": false
    }
  },
  "callLoginParams": {
    "domain": "sip.jitsi.mydomain.com",
    "username": "sip",
    "password": "$PROSODY_SIP_PASSWD"
  }
}
```

### CODE REVIEW

File: src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt [hunk 1]
```diff
@@ -98,17 +98,24 @@ class PjsuaClient(
             command.add("--password=${pjsuaClientParams.sipClientParams.password}")
         }
 
-        if (pjsuaClientParams.sipClientParams.contact != null) {
+        if (!pjsuaClientParams.sipClientParams.contact.isNullOrEmpty()) {
```
Comment (damencho) [inline src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: Is it possible pjsuaClientParams.sipClientParams.contact is null?

File: src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt [hunk 2]
```diff
@@ -98,17 +98,24 @@ class PjsuaClient(
...
             command.add("--contact=${pjsuaClientParams.sipClientParams.contact}")
         }
 
+        if (!pjsuaClientParams.sipClientParams.proxy.isNullOrEmpty()) {
```
Comment (damencho) [inline src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:105]: Same here, is it possible that pjsuaClientParams.sipClientParams.proxy is null?

File: src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt (hunk 1 above)
Comment (emrahcom) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: Yes, if it is not set in the JSON request, its default value is `null` which comes from [SipClient.kt](https://github.com/jitsi/jibri/blob/master/src/main/kotlin/org/jitsi/jibri/sipgateway/SipClient.kt#L23-L53)

Comment (damencho) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: Then that will produce a NPE?

Comment (emrahcom) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: I didn't understand what you meant. They 3 possibilities:

- The `contact` is not in JSON which comes from API request. In this case, it is `null`
- The `contact` is in JSON but its value is an empty string, then it is `""` in the application to
- The `contact` is in JSON and its value is a string (not empty)

Comment (damencho) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: If pjsuaClientParams.sipClientParams.contact is null, then doing `pjsuaClientParams.sipClientParams.contact.isNullOrEmpty()` in kotlin is ok and not producing a NPE? 
In java that will throw null pointer exception, that's why I'm asking.

Comment (emrahcom) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: I also tested for all possible cases in my test environment, with `null`, empty string and non-empty string

Comment (damencho) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: 👍 sorry my kotlin knowledge

Comment (emrahcom) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: It is new for me too :)
But I checked documents

Comment (damencho) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: I also checked but when you told me it is fine :)

Comment (damencho) [reply src/main/kotlin/org/jitsi/jibri/sipgateway/pjsua/PjsuaClient.kt:101]: I should have checked earlier ...
